griptape/memory/structure/conversation_memory.py

```diff
@@ -17,6 +17,7 @@ class ConversationMemory:
     runs: list[Run] = field(factory=list, kw_only=True)
     structure: Structure = field(init=False)
     autoload: bool = field(default=True, kw_only=True)
+    buffer_size: int = field(default=0, kw_only=True)
```

Comment:
I think buffer size of zero should imply that buffer is literally zero, so memory is empty. Let's make it an optional and set the default to `None` instead.

Reply:
Does it make sense to swap out `SummaryConversationMemory.offset` for `buffer_size` to minimize the number of variables we have for memory?

Reply:
@vasinov I set it to `0` because of serialization issues with `None`. But I agree with your point, let me see if I can work around it.

Reply:
Added marshmallow hook to filter out `None`.